tests/mnle_test.py

```diff
@@ -233,43 +263,44 @@ def test_mnle_with_experimental_conditions(mcmc_params_accurate: dict):
     categorical parameter is set to a fixed value (conditioned posterior), and the
     accuracy of the conditioned posterior is tested against the true posterior.
     """
-    num_simulations = 6000
-    num_samples = 500
-
-    def sim_wrapper(theta):
-        # simulate with experiment conditions
-        return mixed_simulator(theta[:, :2], theta[:, 2:] + 1)
+    num_simulations = 10000
+    num_samples = 1000
 
     proposal = MultipleIndependent(
         [
             Gamma(torch.tensor([1.0]), torch.tensor([0.5])),
             Beta(torch.tensor([2.0]), torch.tensor([2.0])),
-            Categorical(probs=torch.ones(1, 3)),
+            BoxUniform(torch.tensor([0.0]), torch.tensor([1.0])),
         ],
         validate_args=False,
     )
 
     theta = proposal.sample((num_simulations,))
-    x = sim_wrapper(theta)
+    x = wrapped_simulator(theta)
     assert x.shape == (num_simulations, 2)
 
     num_trials = 10
-    theta_o = proposal.sample((1,))
-    theta_o[0, 2] = 2.0  # set condition to 2 as in original simulator.
-    x_o = sim_wrapper(theta_o.repeat(num_trials, 1))
+    theta_and_condition = proposal.sample((num_trials,))
+    # use only a single parameter (iid trials)
+    theta_o = theta_and_condition[:1, :2].repeat(num_trials, 1)
+    # but different conditions
+    condition_o = theta_and_condition[:, 2:]
+    theta_and_conditions_o = torch.cat((theta_o, condition_o), dim=1)
+
+    x_o = wrapped_simulator(theta_and_conditions_o)
 
     mcmc_kwargs = dict(
         method="slice_np_vectorized", init_strategy="proposal", **mcmc_params_accurate
     )
 
     # MNLE
-    trainer = MNLE(proposal)
-    estimator = trainer.append_simulations(theta, x).train(training_batch_size=1000)
+    estimator_fun = likelihood_nn(model="mnle", z_score_x=None)
+    trainer = MNLE(proposal, estimator_fun)
+    estimator = trainer.append_simulations(theta, x).train()
 
-    potential_fn = MixedLikelihoodBasedPotential(estimator, proposal, x_o)
-
-    conditioned_potential_fn = ConditionedPotential(
-        potential_fn, condition=theta_o, dims_to_sample=[0, 1]
+    potential_fn, _ = likelihood_estimator_based_potential(estimator, proposal, x_o)
+    conditioned_potential_fn = potential_fn.condition_on(
```

Comment:
Should this be `condition_on_theta`? (this should break and might explain why it is not tested).